### PR TITLE
Api layer

### PR DESCRIPTION
**Pull request by FarhanMobashir**
- adds the api layer to the API provide through context

### CODE REVIEW

File: src/helpers/buildApiHooks.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    let hookName = `use${item.name}`;
+
+    let useQuery = (urlParams = "") => {
+      const [loading, setLoading] = React.useState(true);
+      const [error, setError] = React.useState(false);
+      const [data, setData] = React.useState(null);
+      React.useEffect(() => {
+        console.log("useQuery: ", item.name);
+        setLoading(true);
+        baseQuery(`${item.query}/${urlParams}`)
```
Comment (lannisterrr) [inline src/helpers/buildApiHooks.js:24]: Use of async/await is ideal when you are using es6 syntax.

File: src/helpers/buildApiHooks.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  const hooks = {};
+  queryArray.forEach((item) => {
+    let hookName = `use${item.name}`;
+
+    let useQuery = (urlParams = "") => {
+      const [loading, setLoading] = React.useState(true);
+      const [error, setError] = React.useState(false);
+      const [data, setData] = React.useState(null);
+      React.useEffect(() => {
+        console.log("useQuery: ", item.name);
```
Comment (GuptaEesh) [inline src/helpers/buildApiHooks.js:22]: you can remove these console logs if you are done debugging

**Review by GuptaEesh — COMMENTED** [review]
Code is well-structured.
Great components hierarchy.
Good to merge

**Review by lannisterrr — COMMENTED** [review]
Learnt about useCallback hook , code looks structured , Some suggestions made.

File: src/helpers/buildApiHooks.js (hunk 1 above)
Comment (FarhanMobashir) [reply src/helpers/buildApiHooks.js:24]: will look into that

**Pull request merged**